test/AutoDiff/downstream/inout_parameters.swift

```diff
@@ -125,4 +126,28 @@ InoutParametersTests.test("SetAccessor") {
   expectEqual(8, gradient(at: 4, in: squared))
 }
 
+// Test differentiation wrt `inout` parameters that have a class type.
+InoutParametersTests.test("InoutClassParameter") {
+  class Class: Differentiable {
+    @differentiable
+    var x: Float
+
+    init(_ x: Float) {
+      self.x = x
+    }
+  }
+
+  // Semantically, an empty function with an `inout` parameter is an identity
+  // function.
+  func inoutIdentity(_ c: inout Class) {}
+
+  func identity(_ x: Float) -> Float {
```

Comment:
We need a test case where the `inout`-taking method actually *does* modify the class argument.

Reply:
Done. I added two tests with `inout` class arguments:
- One test assigns a new value to the `inout` class argument (via `store`). This test has correct derivatives.
- The other test mutates the `inout` class argument by applying a property `modify` accessor. This test has incorrect derivatives. TF-1176 tracks the issue, I'd like to address it in a follow-up.

Reply:
> * The other test mutates the `inout` class argument by applying a property `modify` accessor. This test has incorrect derivatives. [TF-1176](https://bugs.swift.org/browse/TF-1176) tracks the issue, it shouldn't be hard to fix but I'd like to address it in a follow-up.

That does not mutate the `inout` class argument. It only mutates the underlying contents, but not the argument itself. To modify the argument, you should assign something to the argument so that the class instance is changed.

Reply:
> To modify the argument, you should assign something to the argument so that the class instance is changed.

I think that's already being done in `squaredViaMutation`. Am I missing sth?
```swift
    func squaredViaMutation(_ c: inout Class) {
      c = Class(c.x * c.x)
    }
    func squared(_ x: Float) -> Float {
      var c = Class(x)
      squaredViaMutation(&c)
      return c.x
    }
```

Reply:
Oh sorry I missed this one. LGTM.